code/datums/materials/_material.dm

```diff
@@ -44,50 +79,97 @@ Simple datum which is instanced once per type and is used for every object of sa
 
 ///This proc is called when the material is added to an object specifically.
 /datum/material/proc/on_applied_obj(obj/o, amount, material_flags)
-	var/new_max_integrity = CEILING(o.max_integrity * integrity_modifier, 1)
-	// This is to keep the same damage relative to the max integrity of the object
-	o.obj_integrity = (o.obj_integrity / o.max_integrity) * new_max_integrity
-	o.max_integrity = new_max_integrity
-	o.force *= strength_modifier
-	o.throwforce *= strength_modifier
+	if(material_flags & MATERIAL_AFFECT_STATISTICS)
+		var/new_max_integrity = CEILING(o.max_integrity * integrity_modifier, 1)
+		// This is to keep the same damage relative to the max integrity of the object
+		o.obj_integrity = (o.obj_integrity / o.max_integrity) * new_max_integrity
+		o.max_integrity = new_max_integrity
+		o.force *= strength_modifier
+		o.throwforce *= strength_modifier
+
+	if(!isitem(o))
+		return
+	var/obj/item/item = o
 
+	if(material_flags & MATERIAL_GREYSCALE)
+		var/worn_path = get_greyscale_config_for(item.greyscale_config_worn)
+		var/lefthand_path = get_greyscale_config_for(item.greyscale_config_inhand_left)
+		var/righthand_path = get_greyscale_config_for(item.greyscale_config_inhand_right)
+		item.set_greyscale(
+			new_worn_config = worn_path,
+			new_inhand_left = lefthand_path,
+			new_inhand_right = righthand_path
+		)
+
+/datum/material/proc/on_applied_turf(turf/T, amount, material_flags)
+	return
+
+/datum/material/proc/get_greyscale_config_for(datum/greyscale_config/config_path)
+	if(!config_path)
+		return
+	for(var/datum/greyscale_config/path as anything in subtypesof(config_path))
+		if(type != initial(path.material_skin))
+			continue
+		return path
 
 ///This proc is called when the material is removed from an object.
-/datum/material/proc/on_removed(atom/source, material_flags)
-	if(!(material_flags & MATERIAL_NO_COLOR)) //Prevent changing things with pre-set colors, to keep colored toolboxes their looks for example
+/datum/material/proc/on_removed(atom/source, amount, material_flags)
+	if(material_flags & MATERIAL_COLOR) //Prevent changing things with pre-set colors, to keep colored toolboxes their looks for example
 		if(color)
 			source.remove_atom_colour(FIXED_COLOUR_PRIORITY, color)
+		if(texture_layer_icon_state)
+//			source.remove_filter("material_texture_[name]")
```

Comment:
Can this filter stuff be added now that we got the plane cube PR?

Reply:
prob